tools/chafa/chafa.c

```diff
@@ -2657,21 +2690,40 @@ run_generic (const gchar *filename, gboolean is_first_file, gboolean is_first_fr
                                         options.font_ratio,
                                         options.scale >= SCALE_MAX - 0.1 ? TRUE : FALSE,
                                         options.stretch);
+            ChafaCanvasConfig * cfg = build_config (
+                dest_width,
+                dest_height,
+                is_animation
+            );
+            ChafaCanvas * canvas = build_canvas (
+                pixel_type, pixels,
+                src_width, src_height, src_rowstride, cfg,
+                placement_id >= 0 ? placement_id + ((frame_count++) % 2) : -1
+            );
 
-            gsa = build_strings (pixel_type, pixels,
-                                 src_width, src_height, src_rowstride,
-                                 dest_width, dest_height,
-                                 is_animation,
-                                 placement_id >= 0 ? placement_id + ((frame_count++) % 2) : -1);
-
-            if (!write_image_prologue (is_first_file, is_first_frame, is_animation, dest_height)
-                || !write_image (gsa, dest_width)
-                || !write_image_epilogue (dest_width)
-                || fflush (stdout) != 0)
-                goto out;
-
-            chafa_free_gstring_array (gsa);
+#ifdef G_OS_WIN32
+            if (options.is_conhost_mode){
+                ConhostRow * lines;
+                gsize s;
+                s=canvas_to_conhost (canvas, &lines);
+                write_image_conhost (lines, s);
+                destroy_lines (lines,s);
+            } else 
+#endif
```

Comment:
GNU style and some spacing. Opening and closing braces `{` and `}` each on its own line, blank line between variable decls and following code. `ConhostRow * lines` should be `ConhostRow *lines`. Spaces around `=`.